ion-schema/src/constraint.rs

```diff
@@ -861,6 +939,11 @@ impl ConstraintValidator for FieldsConstraint {
             let type_def = type_store.get_type_by_id(*type_id).unwrap();
             let values: Vec<&Element> = ion_struct.get_all(field_name).collect();
 
+            // add parent value for current field in ion path
+            ion_path.add_parent(IonPathElement::Field {
```

Comment:
Low-priority observation: you could implement `From<usize>` and `From<&str>` for `IonPathElement` to trim some boilerplate.

```rust
ion_path.push(7);     // IndexedElement
ion_path.push("foo"); // Field